src/lib/recipes.ts

```diff
@@ -274,6 +274,73 @@ const flameAmmoRecipes: IRecipe[] = [
     ]),
 ];
 
+const petrolRecipes: IRecipe[] = [
+    createRecipe([
+        { stuff: Liquids.Oil, count: 100 },
+    ], [
+        { stuff: Liquids.Petrol, count: 75 },
+    ]),
+    createRecipe([
+        { stuff: Liquids.Oil, count: 150 },
+        { stuff: RawResources.Components, count: 5 },
+    ], [
+        { stuff: Liquids.Petrol, count: 120 },
+    ]),
+];
+
+const heavyOilRecipes: IRecipe[] = [
+    createRecipe([
+        { stuff: RawResources.Coal, count: 300 },
+        { stuff: Liquids.Water, count: 100 },
+    ], [
+        { stuff: RawResources.Coke, count: 260 },
+        { stuff: Liquids.HeavyOil, count: 60 },
+    ]),
+    createRecipe([
+        { stuff: Liquids.Oil, count: 200 },
+        { stuff: RawResources.Coke, count: 50 },
+    ], [
+        { stuff: Liquids.HeavyOil, count: 150 },
+    ]),
+];
+
+const enrichedOilRecipes: IRecipe[] = [
+    createRecipe([
+        { stuff: Liquids.HeavyOil, count: 100 },
+        { stuff: RawResources.Components, count: 10 },
+    ], [
+        { stuff: Liquids.EnrichedOil, count: 75 },
+    ]),
+    createRecipe([
+        { stuff: Liquids.HeavyOil, count: 150 },
+        { stuff: RawResources.Coke, count: 75 },
+        { stuff: RawResources.Sulfur, count: 25 },
+    ], [
+        { stuff: Liquids.EnrichedOil, count: 120 },
+    ]),
+];
+
+const oilRecipes: IRecipe[] = [
```

Comment:
Recipes
<img width="729" height="338" alt="image" src="https://github.com/user-attachments/assets/ed826a4a-e3a8-4102-8227-ba853b13177c" />
Use only recipes with materials in input